clang/lib/Interpreter/CodeCompletion.cpp

```diff
@@ -56,26 +69,135 @@ class ReplCompletionConsumer : public CodeCompleteConsumer {
   std::shared_ptr<GlobalCodeCompletionAllocator> CCAllocator;
   CodeCompletionTUInfo CCTUInfo;
   std::vector<std::string> &Results;
+  ReplCodeCompletion &CC;
+};
+
+class CompletionContextHanndler {
```

Comment:
```suggestion
class CompletionContextHandler {
```

Can we add some documentation, too?